.github/workflows/Tests.yaml

```diff
@@ -66,6 +66,22 @@ jobs:
           cd $GITHUB_WORKSPACE
           ./tools/CheckFiles.sh
 
+  # Run format checker for python to make sure the code is pep8 conforming
+  # and passes yapf
+  check_python_formatting:
```

Comment:
You don't need to make this a new job, let's just add a step to the `check_files` job to reduce overhead.

Should we also add a clang-format check?

Reply:
Check files doesn't run in the container. Do you want it to?

No, I was outvoted on that long ago.

Reply:
Since we don't have a clang-format check, we should probably not add a yapf check either.. (don't get me wrong, I'd prefer having both. it would eliminate formatting-only changes in PR diffs)

If we do end up adding the formatting checks, let's run `check_files` in the container and append steps to it to minimize overhead. The git hooks should make sure that the `check_files` job rarely fails anyway.

Reply:
I disagree, I think we should run the yapf check even if we don't run the clangformat check. There's still benefit to be had.

Reply:
ping on this. Please make the `check_files` job run in the container and append the yapf-check step to that job.